client/packages/common/src/ui/discovery/DiscoveredServers.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  AlertIcon,
+  Box,
+  HomeIcon,
+  InlineSpinner,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+  Typography,
+  useTranslation,
+  frontEndHostDisplay,
+  CheckboxEmptyIcon,
+  FrontEndHost,
+  useNativeClient,
+  GqlProvider,
+  QueryClientProvider,
+  QueryClient,
+  frontEndHostGraphql,
+  useInitialisationStatus,
+  InitialisationStatusType,
+} from '@openmsupply-client/common';
+
+type ConnectToServer = ReturnType<typeof useNativeClient>['connectToServer'];
+
+type DiscoverServersProps = {
+  servers: FrontEndHost[];
+  connect: ConnectToServer;
+  discoveryTimedOut: boolean;
+};
+
+export const DiscoveredServers = ({
+  servers,
+  connect,
+  discoveryTimedOut,
+}: DiscoverServersProps) => {
+  const t = useTranslation('app');
+
+  if (servers.length === 0)
+    return (
+      <Box
+        display="flex"
+        flex={1}
+        justifyContent="center"
+        alignContent="center"
+      >
+        <InlineSpinner messageKey="searching" />
+      </Box>
+    );
+
+  if (discoveryTimedOut)
+    return (
+      <Box
+        display="flex"
+        sx={{ color: 'error.main' }}
+        justifyContent="center"
+        flexDirection="column"
+      >
+        <Box display="flex" gap={1}>
+          <Box>
+            <AlertIcon />
+          </Box>
+          <Box>
+            <Typography sx={{ color: 'inherit' }}>
+              {t('error.server-not-found')}
+            </Typography>
+          </Box>
+        </Box>
+      </Box>
+    );
+
+  return (
+    <Box sx={{ minWidth: '300px', color: 'gray.dark' }}>
+      <Typography
+        sx={{
+          fontSize: {
+            xs: '19px',
+            sm: '19px',
+            md: '24px',
+            lg: '32px',
+            xl: '32px',
+          },
+          fontWeight: 700,
+          color: 'primary.main',
+        }}
+      >
+        {t('discovery.select-server')}
+      </Typography>
+      <MenuList style={{ overflowY: 'auto', maxHeight: '200px' }}>
+        {servers.map(server => (
+          <DiscoveredServerWrapper
+            key={frontEndHostGraphql(server)}
+            server={server}
+            connect={connect}
+          />
+        ))}
+      </MenuList>
+    </Box>
+  );
+};
+
+type DiscoveredServerProps = { server: FrontEndHost; connect: ConnectToServer };
+
+const DiscoveredServerWrapper: React.FC<DiscoveredServerProps> = params => (
+  <QueryClientProvider client={new QueryClient()}>
+    <GqlProvider url={frontEndHostGraphql(params.server)}>
+      <DiscoveredServer {...params} />
+    </GqlProvider>
+  </QueryClientProvider>
+);
+
+const DiscoveredServer: React.FC<DiscoveredServerProps> = ({
+  server,
+  connect,
+}) => {
+  const { data: initStatus } = useInitialisationStatus();
+  const t = useTranslation();
+
+  const siteName = () => {
```

Comment:
the usual approach in the f/e codebase is to have a verb in the name if it is a method.. so this would be `getSiteName` otherwise it gets confused with the property `siteName` on `initStatus`

Reply:
Oki, used to follow this patter religiously but moved away from it since in rust the convention for getters to not have `get` prefix, verbalised it: a0b1848387671533faa7d05888c7e1c63dc250be